actions/send_email_workflow_status/action.yml

```diff
@@ -78,16 +81,81 @@ runs:
+          echo "<td>${repo_url}</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>PR</td>" >> ${body_file}
+          echo "<td>${pr_url}</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>PR title</td>" >> ${body_file}
+          echo "<td>${pr_title}</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>PR FROM branch [${branch_source}]</td>" >> ${body_file}
+          echo "<td>PR TO branch [${branch_dest}]</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>PR updated at</td>" >> ${body_file}
+          echo "<td>[${pr_updated_at}]</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>Git merge commit being built</td>" >> ${body_file}
+          echo "<td>${git_sha_merge}</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>Git commit SHA from source branch</td>" >> ${body_file}
+          echo "<td>${git_sha_source}</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>Workflow run url</td>" >> ${body_file}
+          echo "<td>${run_url}</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>Build module</td>" >> ${body_file}
+          echo "<td>$(sed -n '1p' < ./version.txt)</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+          echo "<tr>" >> ${body_file}
+          echo "<td>Build version</td>" >> ${body_file}
+          echo "<td>$(sed -n '2p' < ./version.txt)</td>" >> ${body_file}
+          echo "</tr>" >> ${body_file}
+
+           # if docker.txt exists, grab docker registry / tags of module and include in email
+          if [ -f ./docker.txt ]; then
+            echo "<tr>" >> ${body_file}
+            echo "<td>Docker registry project</td>" >> ${body_file}
+            echo "<td>$(sed -n '1p' < ./docker.txt)</td>" >> ${body_file}
+            echo "</tr>" >> ${body_file}
+
+            echo "<tr>" >> ${body_file}
+            echo "<td>Docker registry tags</td>" >> ${body_file}
+            echo "<td>$(sed -n '2p' < ./docker.txt)</td>" >> ${body_file}
+            echo "</tr>" >> ${body_file}
+          fi
+          echo "</table>" >> ${body_file}
+
+          echo "<br>" >> ${body_file}
+          echo "<br>" >> ${body_file}
+
+          # replace all /n with <br>
+          sed -i '{:q;N;s/\n/<br>/g;t q}' ./mvn_results.txt
+
+          # append mvn_results.txt, which is the mvn command output to body_file
+          cat mvn_results.txt >> ${body_file} 
+
         elif [[ "push" == "${event_name}" ]]; then
```

Comment:
Hi,
Some blocs a common to the 2 cases (pull request VS push) : Build module and docker part.
Maybe we could have only different parts using if, something like:
```
...common part...
if pull_request
  specific part for PR
elif
  specific part for push
... again common part ...
```

Also we could use a templating engine. The only GitHub action I found is using jinja:
https://github.com/cuchi/jinja2-action
https://github.com/marketplace/actions/jinja2-action
But it would add more complexity, so probably for such simple case we can live with some hardcoded html pieces.

Reply:
I removed few items from pull_request if statement as they were not being used
Moved duplicated section to end of bash script after if / elff /else